### ISSUE TITLE
No format pattern was found for unit ... with grammatical case :nominative, gender :neuter and plural type :one

### ISSUE THREAD

**Issue by ghost** [kipcole9/cldr_units#21]
👋

I've just upgraded to the latest ExCLDR versions, and got the following error in one of the tests (which checks that all measurements used in the app can be "rendered"):

```elixir
iex(1)> App.Cldr.Unit.to_string!(1, unit: :pascal, style: :short)
** (Cldr.Unit.NoPatternError) No format pattern was found for unit #Cldr.Unit<:pascal, 1> with grammatical case :nominative, gender :neuter and plural type :one
    (ex_cldr_units 3.5.1) lib/cldr/unit/format.ex:675: Cldr.Unit.Format.get_unit_pattern!/6
    (ex_cldr_units 3.5.1) lib/cldr/unit/format.ex:403: Cldr.Unit.Format.to_iolist/3
    (ex_cldr_units 3.5.1) lib/cldr/unit/format.ex:194: Cldr.Unit.Format.to_string/3
    (ex_cldr_units 3.5.1) lib/cldr/unit/format.ex:281: Cldr.Unit.Format.to_string!/3
```

Without `:style` option it works as expected:

```elixir
iex(1)> App.Cldr.Unit.to_string!(1, unit: :pascal)               
"1 Pascal"
```

I wonder if maybe `:style` has been replaced with some other option?

Versions:

```
"ex_cldr": "2.20.0"
"ex_cldr_units": "3.5.1"
```

Thank you!

**Comment by kipcole9** [comment]
Thanks for the report and sorry for the inconvenience. I've just tried on master branch and it seems ok. Can you let me know what locale you are operating in? Perhaps that makes a difference since not all locales have all data for units yet (its a CLDR work in progress).

The error indicates that `ex_cldr_units` couldn't find a unit pattern for 
```elixir
iex> Cldr.Unit.to_string!(1, [unit: :pascal, style: :short])
"1 Pa"
iex> MyApp.Cldr.Unit.to_string!(1, [unit: :pascal, style: :short])
"1 Pa"
```

**Comment by ghost** [comment]
Ah, right, sorry, forgot to mention that the default locale is set to `de`.

```elixir
iex(3)> App.Cldr.Unit.to_string!(1, [unit: "pascal", style: :narrow, locale: "en"])
"1Pa"
iex(4)> App.Cldr.Unit.to_string!(1, [unit: "pascal", style: :narrow, locale: "de"])
** (Cldr.Unit.NoPatternError) No format pattern was found for unit #Cldr.Unit<:pascal, 1> with grammatical case :nominative, gender :neuter and plural type :one
    (ex_cldr_units 3.5.1) lib/cldr/unit/format.ex:675: Cldr.Unit.Format.get_unit_pattern!/6
    (ex_cldr_units 3.5.1) lib/cldr/unit/format.ex:403: Cldr.Unit.Format.to_iolist/3
    (ex_cldr_units 3.5.1) lib/cldr/unit/format.ex:194: Cldr.Unit.Format.to_string/3
    (ex_cldr_units 3.5.1) lib/cldr/unit/format.ex:281: Cldr.Unit.Format.to_string!/3
```

**Comment by kipcole9** [comment]
Ahhhh, I suspect you are in the "de" locale given it refers to the `:neuter` gender. And only locales "de" and "fr" currently have grammatical features in them.

**Comment by kipcole9** [comment]
Confirmed and can reproduce. I will explore further what I can do to mitigate the data gap. Should be able to resolve during your day and I'll publish an update when done.

**Comment by ghost** [comment]
Thank you!

**Comment by kipcole9** [comment]
I have published [ex_cldr_units 3.5.2](https://hex.pm/packages/ex_cldr_units/3.5.2) that fixes the error with [this commit](https://github.com/elixir-cldr/cldr_units/commit/a9111ffd6cdbe80caac0532f8c1ac8712b9be891).  The changelog entry is:

#### Bug Fixes

* Fix getting a unit pattern when the unit value is zero, one or two and there is no pattern for the default unit plural category. Thanks to @syfgkjasdkn for the report.  Closes #21.

Thanks for your patience and collaboration!
